ko/sql/tuning.rst

```diff
@@ -3831,6 +3831,92 @@ N:1 관계의 **LEFT OUTER JOIN**\에서 조인 조건 외에 오른쪽 테이
                 9           60          600  'Left- 806'                     6
                10           60          600  'Left- 906'                     6
 
+.. _pred_push
+
+Predicate Push
+-----------------------
+Predicate Push는 뷰를 조회하기 전에 먼저 조건절을 적용한다. 
+
+이를 통해 조건에 만족하는 더 적은 양의 데이터만 조회되도록 하여 전체 처리량을 줄일 수 있다.
+
+
+예를 들어 **GROUP BY**\절을 포함한 아래 뷰를 처리할 때, 쿼리 블록 밖에 있는 조건절을 쿼리 블록 안쪽에 밀어 넣을 수 있다면
+**GROUP BY** 해야 할 데이터양을 줄일 수 있다. 
+
+
+.. code-block:: sql
+
+
+        SELECT a.name, r.score 
+        FROM (SELECT * FROM athlete) a
+        WHERE a.code = r.athlete_code
+        AND a.nation_code = 'KOR';
+
+
+위 쿼리에 정의한 뷰 내부에는 조건절이 없다. 만약 쿼리 변환이 작동하지 않는다면,
+*emp* 테이블을 Full Scan 하고서 **GROUP BY** 이후에 *deptno = 30* 조건을
+필터링했을 것이다. 
+
+하지만 Predicate Push를 통해서 다음과 같이 쿼리가 변환된다면, 더 적은 양의
+데이터만 조회되도록 최적화할 수 있다.
+
+
+.. code-block:: sql
+
+
+        SELECT a.name, r.score 
+        FROM (SELECT * FROM athlete WHERE a.nation_code = 'KOR') a
+        WHERE a.code = r.athlete_code
+        AND a.nation_code = 'KOR';
+
+다음의 경우엔 Predicate Push가 수행되지 않는다.
+
+    #. **CONNECT BY**\를 포함한 경우
+
+    #. 집계함수나 분석함수를 사용하는 경우
+
+    #. **ROWNUM, LIMIT** 또는 **GROUPBY_NUM (), INST_NUM (), ORDERBY_NUM ()**\ 을 사용하는 경우
+
+    #. **Correlated Subquery**\ 를 사용하여 작성된 경우
+
+    #. **OUTER JOIN**\의 **ON**\절에 조건절이 작성된 경우
+
+    #. 조건절에 부질의가 사용된 경우
+
+    #. 뷰가 메소드를 포함한 경우
+
+    #. OUTER JOIN을 수행할 때 푸시될 조건절이나 뷰 내부의 Predicate Push 대상에 NULL 변환 함수를 사용할 경우
+
+    #. OUTER JOIN을 수행할 때 푸시될 조건절이나 뷰 내부의 Predicate Push 대상에 부질의를 사용하는 경우
+
+    #. OUTER JOIN을 수행할 때 푸시될 조건절이나 뷰 내부의 Predicate Push 대상에 **RANDOM (), DRANDOM (), SYS_GUID ()**\를 사용하는 경우
+
+다음은 질의가 OUTER JOIN을 수행할 때 **ON**\절의 조건에 푸시될 조건절이 있는 예시이다.
+
+.. code-block:: sql
+
+        SELECT a.name, r.score 
+        FROM (SELECT * FROM athlete) a
+        LEFT OUTER JOIN record r 
+        ON a.code = r.athlete_code
+        AND a.nation_code = 'KOR';
+
+이 경우, a.nation_code = 'KOR'는 LEFT OUTER JOIN 수행 시 ON 절에 있는데, 이러한 형태로 ON 절의 조건절은 Predicate Push 대상이 아니다.
+
+다음 질의는 OUTER JOIN을 수행할 때 푸시될 조건절이나 뷰 내부의 Predicate Push 대상에 NULL 변환 함수를 사용하는 예시이다.
+
+.. code-block:: sql
+
+        SELECT a.name, r.score 
+        FROM athlete a
+        LEFT OUTER JOIN (SELECT * FROM record) r 
```

Comment:
This example results in view merging before even considering predicate push.
Even if the /*+ no_merge */ hint is used, views are merged.
So please consider adding the predicate below.

```suggestion
        LEFT OUTER JOIN (SELECT * FROM record WHERE medal = 'G') r 
```